Subject: share/gprconfig/*.xml fixes for Debian
 Patch all files in share/gprconfig/
 .
 Require the default compilers to bo version 4.9.  This ensures
 compatibility between objects compiled from different languages.
 .
 Prefer gnatgcc to gcc for Ada. This requires a work-around in gprlib,
 because it does strong assertions about the compiler name.
 .
 We put gprbind and gprlib in /lib/gprbuild, not /libexec/gprbuild. We don't
 put them in /usr/lib/${DEB_HOST_MULTIARCH}/gprbuild, because the multiarch
 conventions do not yet address executables.
 Moreover, their path must be written in the arch-indep
 /usr/share/gprconfig/*.xml.
Author: Stephen Leake <stephen_leake@stephe-leake.org>
Author: Nicolas Boulenguez <nicolas@debian.org>
Forwarded: not-needed

--- a/share/gprconfig/compilers.xml
+++ b/share/gprconfig/compilers.xml
@@ -12,9 +12,8 @@
     <name>GCC</name>
     <executable prefix="1">(.*(-wrs-|-sysgo|-elf-|-eabi-|-eabispe-|avr-|-elinos-linux|-pc-mingw32|-xcoff-|arm-linux-androideabi-|arm-linux-gnueabi-).*)?gcc</executable>
     <version>
-      <external>${PREFIX}gcc -v</external>
-      <nogrep regexp="for GNAT Pro"></nogrep>
-      <grep regexp="^gcc \S+ (\S+)" group="1"></grep>
+      <external>${PREFIX}gcc -dumpversion</external>
+      <grep regexp="(4\.9)(\.\d+)?" group="1"></grep>
     </version>
     <languages>C</languages>
     <target>
@@ -312,9 +311,8 @@
     <name>G++</name>
     <executable>g\+\+</executable>
     <version>
-      <external>g++ -v</external>
-      <nogrep regexp="for GNAT Pro"></nogrep>
-      <grep regexp="^gcc \S+ (\S+)" group="1"></grep>
+      <external>${PREFIX}g++ -dumpversion</external>
+      <grep regexp="(4\.9)(\.\d+)?" group="1"></grep>
     </version>
     <languages>C++</languages>
     <target>
@@ -632,8 +630,8 @@
     <name>GCC-ASM</name>
     <executable prefix="1">(.*(-wrs-|-elf-|-eabi-|-eabispe-|avr-|elinos-linux|-pc-mingw32).*)?gcc</executable>
     <version>
-      <external>${PREFIX}gcc -v</external>
-      <grep regexp="^gcc \S+ (\S+)" group="1"></grep>
+      <external>${PREFIX}gcc -dumpversion</external>
+      <grep regexp="(4\.9)(\.\d+)?" group="1"></grep>
     </version>
     <languages>Asm,Asm_Cpp,Asm2</languages>
     <target>
@@ -652,21 +650,18 @@
     <name>GNAT</name>
     <executable prefix="1">(.*(-wrs-|-sysgo|-elf-|-eabi-|-eabispe-|avr-|elinos-linux|-pc-mingw32|-xcoff-|-pikeos-|arm-linux-androideabi-|arm-linux-gnueabi-).*)?gnatmake</executable>
     <version>
-      <external>${PREFIX}gnatls -v</external>
-      <grep regexp="^GNATLS.+?(\d+(\.\d+)?)" group="1"></grep>
+      <external>${PREFIX}gcc -dumpversion</external>
+      <grep regexp="(4\.9)(\.\d+)?" group="1"></grep>
     </version>
     <languages>Ada</languages>
-    <variable name="gcc_version">
-      <external>${PREFIX}gcc -v</external>
-      <grep regexp="^[-\w]*gcc \S+ (\S+)" group="1"></grep>
-    </variable>
+    <variable name="gcc_version">4.9</variable>
     <runtimes default="default,kernel,native">
        <directory group="default" >\.\./lib/gcc(-lib)?/$TARGET/$gcc_version/adalib/</directory>
        <directory group="default" contents="^rts-">\.\./lib/gcc(-lib)?/$TARGET/$gcc_version/ada_object_path</directory>
        <directory group="2" >\.\./lib/gcc(-lib)?/$TARGET/$gcc_version/rts-(.*)/adalib/</directory>
     </runtimes>
     <target>
-      <external>${PREFIX}gcc -dumpmachine</external>
+      <external>${PREFIX}gcc -dumpmachine</external>
       <grep regexp="[^\r\n]+"></grep>
     </target>
   </compiler_description>
@@ -728,8 +723,8 @@
     <name>GFORTRAN</name>
     <executable>gfortran</executable>
     <version>
-      <external>gfortran -v</external>
-      <grep regexp="^gcc \S+ (\S+)" group="1"></grep>
+      <external>${PREFIX}gfortran -dumpversion</external>
+      <grep regexp="(4\.9)(\.\d+)?" group="1"></grep>
     </version>
     <languages>Fortran</languages>
     <target>
